src/main/config.cpp

```diff
@@ -0,0 +1,84 @@
+#include <iostream>
+#include <sstream>
+#include "gflags/gflags.h"
+#include "config.hpp"
+
+extern "C" {
+	#include "exec_monitor.h"
+}
+
+/*
+	FLAGS SECTION
+*/
+
+// EXEC FLAG - MONITOR PROCESS EXECUTION
+DEFINE_bool(exec, false, "Monitors process executions.");
+DEFINE_string(ppid, "", "Specifies the ppid(s) of the processes to monitor using comma-separated values.");
+DEFINE_string(name, "", "Specifies the name(s) of the processes to monitor using comma-separated values.");
+DEFINE_int32(n, 0, "Specifies the number of processes to monitor. The program will stop after n processes were executed.");
+DEFINE_int32(t, 0, "Specifies the number of seconds to monitor. The program will stop after t seconds.");
+
+// DATA EXPORT FLAGS
+DEFINE_string(save, "", "Specifies the format in which to save the terminal output in a separate file. Available formats are csv and protobuf.");
+DEFINE_validator(save, Config::checkSaveAndExportFormat);
+DEFINE_string(export, "", "Specifies the format in which to export data in a separate file. Using export will not print to stdout. Available formats are csv and protobuf.");
+DEFINE_validator(export, Config::checkSaveAndExportFormat);
```

Comment:
How do you imagine those flags to interact? Are they mutually exclusive ? (this is my understanding of "Using export will not print to stdout"? If that's the case, you could have only one flag "format" that is either csv or protobuf and you could have another flag, "output_file" that, if used, would tell hawk to save data to disk (and not print).

If your intention is to have them work in parallel (ie have hawk both print stuff and write data to disk) maybe you should think about the flag that will contain the output file and rename those so that it's clearer that they are about the output format (maybe save_format or save_fmt ?)

Reply:
I was actually thinking that there would be 3 output options: just in the terminal, just to a file (csv/binary) or both (terminal and file at the same time; terminal to see in real time and file so you can use the data in another program). Now that I think of it more, I don't think it's necessary to accustom both at the same time, so if you want to export to a file, then it's fine not to print on the terminal anymore. 

So I would use a flag, --format, which can take values such as csv and protobuf, and if it's used, it will export the data to a file and automatically not print it to stdout. If it's not specified, it will print to stdout. How does this sound?